CHANGELOG.md

```diff
@@ -29,6 +29,7 @@ The following emojis are used to highlight certain changes:
 
 - 🛠 `boxo/gateway`: when making a trustless CAR request with the "entity-bytes" parameter, using a negative index greater than the underlying entity length could trigger reading more data than intended
 - 🛠 `boxo/gateway`: the header configuration `Config.Headers` and `AddAccessControlHeaders` has been replaced by the new middleware provided by `NewHeaders`.
+- 🛠 `routing/http/client`: a potential race condition has been fixed when creating multiple clients within the same process. With this change, `WithHTTPClient` and `WithUserAgent` may not always work with each other. More details can be found in the documentation.
```

Comment:
> With this change, `WithHTTPClient` and `WithUserAgent` may not always work with each other. More details can be found in the documentation.

Wasn't this always the case? If you called `WithHTTPClient` and used a non-compatible client `WithUserAgent` would be silently ignored. AFAICT the changes are:
1. Incompatible options will now return errors rather than silently failing
2. The default HTTP client is no longer a singleton and therefore passing `WithUserAgent` won't modify the user agent of existing routing clients with default http clients.